specifyweb/frontend/js_src/lib/components/Forms/DeleteButton.tsx

```diff
@@ -77,16 +79,27 @@ export function DeleteButton<SCHEMA extends AnySchema>({
 
   return (
     <>
-      <ButtonComponent
-        title={isBlocked ? formsText.deleteBlocked() : undefined}
-        onClick={(): void => {
-          handleOpen();
-          setDeferred(false);
-        }}
-      >
-        {isBlocked ? icons.exclamation : undefined}
-        {commonText.delete()}
-      </ButtonComponent>
+      {isIcon ? (
+        <Button.Icon
+          title={isBlocked ? formsText.deleteBlocked() : commonText.delete()}
+          icon="trash"
+          onClick={(): void => {
```

Comment:
(optional) can you extract the callback into a common function instead of duplicating it two times?